src/Validate/Weight/CartWeightValidator.php

```diff
@@ -3,16 +3,73 @@
 namespace Invertus\dpdBaltics\Validate\Weight;
 
 use Invertus\dpdBaltics\Config\Config;
+use PrestaShop\PrestaShop\Adapter\Validate;
 
 class CartWeightValidator
 {
-    public function validate($cartWeight, $countryIso, $productReference)
+    public function validate($cart, $countryIso, $productReference)
     {
+        $parcelDistribution = \Configuration::get(Config::PARCEL_DISTRIBUTION);
         $maxAllowedWeight = Config::getDefaultServiceWeights($countryIso, $productReference);
+
         if (!$maxAllowedWeight) {
             return true;
         }
 
-        return $maxAllowedWeight >= $cartWeight;
+        if (!Validate::isLoadedObject($cart)) {
+            return false;
+        }
+
+        switch ($parcelDistribution) {
+            case \DPDParcel::DISTRIBUTION_NONE:
+                return $maxAllowedWeight >= $cart->getTotalWeight();
+            case \DPDParcel::DISTRIBUTION_PARCEL_QUANTITY:
+                return $this->isProductQuantityDistributionWeightValid($cart, $maxAllowedWeight);
+            case \DPDParcel::DISTRIBUTION_PARCEL_PRODUCT:
+                return $this->isProductDistributionWeightValid($cart, $maxAllowedWeight);
+            default :
+                return true;
```

Comment:
you sure you want to validate it if no switch case was found?

Reply:
yeah possibly an exception or if you want silent kill then atleast log error quitely. And safer would be to return false ?